src/menus/DirektverbindungenMenu/DvLayerSwitcher.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable no-param-reassign */
+import React, { useEffect, useState, useMemo, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import { FormGroup, FormControlLabel, makeStyles } from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+import { unByKey } from 'ol/Observable';
+import SBBSwitch from '../../components/SBBSwitch';
+import { DV_DAY_NIGHT_REGEX } from '../../utils/constants';
+import { setFeatureInfo } from '../../model/app/actions';
+
+const useStyles = makeStyles(() => {
+  return {
+    switchWrapper: {
+      '& .MuiFormControlLabel-label': {
+        fontSize: 16,
+      },
+      '& .MuiFormControlLabel-root': {
+        minWidth: 155,
+      },
+      '& .MuiSwitch-switchBase': {
+        top: (props) => (props.isMobile && props.row ? 4 : 0),
+      },
+    },
+    disabled: {
+      cursor: 'not-allowed',
+      '& .MuiButtonBase-root': {
+        cursor: 'not-allowed',
+      },
+      '& .MuiSwitch-root': {
+        opacity: 0.6,
+      },
+    },
+  };
+});
+
+function DvLayerSwitcher({ onToggle, row }) {
+  const { t } = useTranslation();
+  const [revision, forceRender] = useState();
+  const dispatch = useDispatch();
+  const layers = useSelector((state) => state.map.layers);
+  const screenWidth = useSelector((state) => state.app.screenWidth);
+  const featureInfo = useSelector((state) => state.app.featureInfo);
+  const isMobile = useMemo(() => {
+    return ['xs'].includes(screenWidth);
+  }, [screenWidth]);
+  const classes = useStyles({ isMobile, row });
+  const dvLayers = useMemo(
+    () =>
+      layers.filter((layer) => {
+        return DV_DAY_NIGHT_REGEX.test(layer.key);
+      }),
+    [layers],
+  );
+  const getVisibleLayerKeys = useCallback(
+    () => dvLayers.filter((l) => l.visible).map((l) => l.key),
+    [dvLayers],
+  );
+  const [layersVisible, setLayersVisible] = useState(getVisibleLayerKeys());
+
+  // Force render when visibility changes
+  useEffect(() => {
+    const olKeys =
+      layers?.map((layer) => {
+        return layer?.on('change:visible', (evt) => {
+          forceRender(revision + 1);
+          const { target: targetLayer } = evt;
+          if (DV_DAY_NIGHT_REGEX.test(targetLayer.key)) {
+            setLayersVisible(getVisibleLayerKeys());
```

Comment:
Same comment as  in other layerSwitcher you can define layersVisible outside useXXX functions.  It will simplify the code. or use a useMemo when  revision updates

Reply:
Removed from state